Objects/funcobject.c

```diff
@@ -19,9 +19,21 @@ PyFunction_NewWithQualName(PyObject *code, PyObject *globals, PyObject *qualname
             return NULL;
     }
 
+    /* __module__: If module name is in globals, use it.
+    Otherwise, use None. */
+    module = PyDict_GetItemWithError(globals, __name__);
```

Comment:
This now runs before `Py_INCREF(globals)`, not sure if relevant (because our caller should hold a reference to `globals`, right?)

Can put `Py_INCREF(globals)` before this call.